packages/cardano-services/test/entrypoints.test.ts

```diff
@@ -24,6 +29,13 @@ const assertServiceHealthy = async (apiUrl: string, serviceName: ServiceNames) =
   expect(res.data).toEqual({ ok: true });
 };
 
+const getCommandLineOptionsWithoutDb = (serviceName: ServiceNames) => [
```

Comment:
This is a step in the right direction, but it's not reducing duplication that much, and it leaves the tests a little awkward to read. Try this:

```ts
const callCliAndAssertExit = (
  { args, env = {}, expectedExitCode }: { args: string[]; env?: NodeJS.ProcessEnv; expectedExitCode: number },
  done: jest.DoneCallback
) => {
  const spy = jest.fn();
  expect.assertions(2);
  const proc = fork(exePath('cli'), ['start-server', '--logger-min-severity', 'error', ...args], {
    env,
    stdio: 'pipe'
  });
  proc.stderr!.on('data', spy);
  proc.on('exit', (code) => {
    expect(code).toBe(expectedExitCode);
    expect(spy).toHaveBeenCalled();
    done();
  });
};
... 
         it('stake-pool exits with code 1', (done) => {
            callCliAndAssertExit({
              args: ['--services', ServiceNames.StakePool],
              expectedExitCode: 1
            },
            done);
          });
```

Reply:
Tests optimization is applied under 39834e029599adf5a77143c3fdb759ba46e36976